src/beta_inc.jl

```diff
@@ -0,0 +1,815 @@
+    t = inv(b)^2
+    w = @horner(t, .833333333333333E-01, -.277777777760991E-02*s₃, .793650666825390E-03*s₅, -.595202931351870E-03*s₇, .837308034031215E-03*s₉, -.165322962780713E-02*s₁₁)
+    w *= c/b
+    # COMPUTE stirling(a) + w
+    t = inv(a)^2
+    return @horner(t, .833333333333333E-01, -.277777777760991E-02, .793650666825390E-03, -.595202931351870E-03, .837308034031215E-03, -.165322962780713E-02)/a + w
+end
+
+"""
+    esum(mu,x)
+
+Compute ``e^{mu+x}``
+"""
+function esum(mu::Float64, x::Float64)
+    if x > 0.0
+        if mu > 0.0 || mu + x < 0.0
+            return exp(mu)*exp(x)
+        else
+            return exp(mu + x)
+        end
+    elseif mu < 0.0 || mu + x > 0.0
+        return exp(mu)*exp(x)
+    else
+        return exp(mu + x)
+    end
+end
+
+"""
+    brcmp1(mu,a,b,x,y,case)
+
+If case == 1: compute ``e^{mu} * x^{a}y^{b}/B(a,b)``
+else : compute above with mu = 0.0
+"""
+function brcmp1(mu::Float64,a::Float64,b::Float64,x::Float64,y::Float64,case::Bool)
+    a0, b0 = minmax(a,b)
+    if a0 >= 8.0
+        if a > b 
+            h = b/a
+            x0 = 1.0/(1.0 + h)
+            y0 = h/(1.0 + h)
+            lambda = (a+b)*y - b
+        else
+             h = a/b
+             x0 = h/(1.0 + h)
+             y0 = 1.0/(1.0 + h)
+             lambda = a - (a+b)*x
+        end
+        e = -lambda/a
+        u = abs(e) > 0.6 ? u = e - log(x/x0) : - log1pmx(e)
+        e = lambda/b
+        v = abs(e) > 0.6 ? e - log(y/y0) : - log1pmx(e)
+        z = case ? esum(mu, -(a*u + b*v)) : exp(-(a*u+b*v))
+        return (1.0/sqrt(2*pi))*sqrt(b*x0)*z*exp(-bcorr(a,b))
+    elseif x > 0.375
+        if y > 0.375
+            lnx = log(x)
+            lny = log(y)
+        else
+            lnx = log1p(-y)
+            lny = log(y)
+        end
+    else
+        lnx = log(x)
+        lny = log1p(-x)
+    end
+    z = a*lnx + b*lny
+    if a0 < 1.0
+        b0 = max(a,b)
+        if b0 >= 8.0
+            u = loggamma1p(a0) + loggammadiv(a0,b0)
+            return a0*(case ? esum(mu, z-u) : exp(z-u))
+        elseif b0 > 1.0
+            u = loggamma1p(a0)
+            n = trunc(Int,b0 - 1.0)
+            if n >= 1
+                c = 1.0
+                for i = 1:n
+                    b0 -= 1.0
+                    c *= (b0/(a0+b0))
+                end
+                u += log(c)
+            end
+            z -= u
+            b0 -= 1.0
+            apb = a0 + b0
+            if apb > 1.0
+                u = a0 + b0 - 1.0
+                t = (1.0 + rgamma1pm1(u))/apb
+            else
+                t = 1.0 + rgamma1pm1(apb)
+            end
+            return a0*(case ? esum(mu,z) : exp(z))*(1.0 + rgamma1pm1(b0))/t
+        end
+    else
+        z -= logbeta(a,b)
+        ans = case ? esum(mu,z) : exp(z)
+        return ans
+    end
+
+   
+    
+    if ans == 0.0
+        return 0.0
+    end
+    apb = a + b
+    if apb > 1.0
+        u = a + b - 1.0
+        z = (1.0 + rgamma1pm1(u))/apb
```

Comment:
```suggestion
        z = (1.0 + rgamma1pm1(apb - 1))/apb
```
and delete the line above